### PR TITLE
Add comprehensive tests to fromHex/toHex

### PR DESCRIPTION
**Pull request by axic**
Follow up of #7785 and #7786.

### CODE REVIEW

**Review by chriseth — APPROVED** [review]

File: test/libdevcore/CommonData.cpp [hunk 1]
```diff
@@ -28,16 +28,72 @@
 using namespace std;
 using namespace dev::solidity;
 
+// TODO: Fix Boost...
+BOOST_TEST_DONT_PRINT_LOG_VALUE(dev::bytes);
```
Comment (ekpyron) [inline test/libdevcore/CommonData.cpp:32]: Maybe define this helper and use it instead?
```
#define CHECK_EQUAL_RANGE(x,y) do { auto const& lhs = (x); auto const& rhs = (y); BOOST_CHECK_EQUAL_COLLECTIONS(lhs.begin(), lhs.end(), rhs.begin(), rhs.end()); } while(0)
```

Comment (chriseth) [reply test/libdevcore/CommonData.cpp:32]: The error message output in that case is not really helpful.

Comment (ekpyron) [reply test/libdevcore/CommonData.cpp:32]: But the current hack is also fine I guess...

Comment (axic) [reply test/libdevcore/CommonData.cpp:32]: Well could just use `BOOST_CHECK` and manually print also..

**Pull request merged**